src/operations/debugProbeFirmwares.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
+ */
+
+import { readdirSync, readSync, openSync, closeSync } from 'node:fs';
+import { join } from 'node:path';
+
+interface HostFirmwareInfo {
+    filePath: string;
+    compiledOn: Date;
+}
+
+interface HostFirmwares {
+    [key: string]: HostFirmwareInfo;
+}
+
+interface NrfUtilDevice {
+    deviceInfo: { jlink: { jlinkObFirmwareVersion: string } };
+}
+
+/**
+ * Reads the debug probe firmware files in the given JLink directory and returns
+ * a map of their IDs and compilation dates.
+ */
+export const getHostFirmwareVersions = (jlinkDir: string): HostFirmwares => {
+    const buffer = Buffer.alloc(64);
+
+    return Object.fromEntries(
+        readdirSync(join(jlinkDir, 'Firmwares'))
+            .map(file => {
+                const filePath = join(jlinkDir, 'Firmwares', file);
+                const fd = openSync(filePath, 'r');
+                readSync(fd, buffer, 0, buffer.length, 0);
+                const [id, dateStr] = buffer
+                    .toString()
+                    .split('\0')[0]!
+                    .split(' compiled ');
+                closeSync(fd);
+                return id && dateStr
+                    ? [
+                          id,
+                          <HostFirmwareInfo>{
+                              filePath,
+                              compiledOn: new Date(dateStr),
+                          },
+                      ]
+                    : undefined;
```

Comment:
I cannot really judge whether it is reliable that the J-Link firmwares always contain these strings in this format at the beginning. I assume you know about these things better than I do.

Reply:
> I cannot really judge whether it is reliable that the J-Link firmwares always contain these strings in this format at the beginning. I assume you know about these things better than I do.

This is why the system team is refusing to do this for us. But regardless if SEGGER breaks this at any point, until then it is still useful. As this information is in the firmwares for as long as you can trace it back, it is quite unlikely that it will ever change. If it does most likely there will be an API to access this.